modules/Bio/EnsEMBL/Compara/RunnableDB/GeneTrees/OrthoTree.pm

```diff
@@ -538,8 +567,15 @@ sub store_gene_link_as_homology {
         $self->param('orthotree_homology_counts')->{'gene_split'}++;
     }
   }
-  
-  $self->param('homologyDBA')->store($homology) unless $self->param('_readonly');
+
+    if ( $self->param('output_flatfile') ) {
```

Comment:
It's strange to add `$output_fh` to so many function calls and in the end use `$self->param('output_flatfile')` to check whether to use it or not.
Either use `$output_fh` in the `if` here, or put `$output_fh` in `$self->param` when the file is opened and you don't need to modify the signatures of `tag_genepairlink` and `store_gene_link_as_homology`